mempool/src/core_mempool/transaction.rs

```diff
@@ -84,6 +89,13 @@ pub struct SequenceInfo {
     pub account_sequence_number: u64,
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct InsertionInfo {
+    pub insertion_time: SystemTime,
+    pub client_submitted: bool,
+    pub is_end_to_end: bool,
```

Comment:
nit: can we add a comment to this line? I'm a noob and I'm not sure what `is_end_to_end` means? 🤔

Reply:
Good call. I renamed is_end_to_end and added a comment to both of those, as unfortunately they're quite nuanced.